Fix monkey patch of ActiveRecord::Base for records without primary key

atranson-electra opened the pull request:
## Context

On the project I am working on, we are relying on TimescaleDB PostgreSQL extension.
The corresponding hypertables and record implementation does not expose a `primary_key` attribute on the record class nor does it have an `id` at the instance level.

This means that all calls to `attributes_for_super_diff` result with `NoMethodError: undefined method `to_sym' for nil`.

## Fix

This PR proposes to edit the monkey patch to handle this specific case.

atranson-electra committed:
Fix monkey patch of ActiveRecord::Base for records without primary keys

jas14 reviewed (changes requested):
Thanks, @atranson-electra ! Please add a [test](https://github.com/splitwise/super_diff/blob/88a123cfbbc6a16694f1cf1ee3e9a2cb0eba3b42/spec/unit/active_record/object_inspection_spec.rb) – other than that, looks great.

If you're feeling inspired, there are [other](https://github.com/splitwise/super_diff/blob/25ceb5f01dfecc1c75f45958278a38a62233de36/lib/super_diff/active_record/operation_tree_builders/active_record_model.rb#L15) [parts](https://github.com/splitwise/super_diff/blob/25ceb5f01dfecc1c75f45958278a38a62233de36/lib/super_diff/active_record/inspection_tree_builders/active_record_model.rb#L12) of the code base that rely on the presence of a `primary_key`. Feel free to address those in a separate PR, or to leave them for someone else.

atranson-electra committed:
Fix usage of nil primary key in tree builders related to active records

atranson-electra committed:
Add tests to validate output for active records without primary keys

atranson-electra committed:
Update changelog to highlight new fixes

atranson-electra commented:
@jas14 I updated the other parts using the primary key and added test on the way :+1:

I believe this covers it.
With the extra changes, diff reports of records without primary key won't have `: nil` and `id: nil` strings hanging around anymore.

jas14 reviewed (approved):
Fantastic!
